projects/norbot_systems_identification/angular_velocity.cc

```diff
@@ -0,0 +1,130 @@
+// BoB robotics includes
+#include "common/assert.h"
+#include "common/background_exception_catcher.h"
+#include "common/main.h"
+#include "common/stopwatch.h"
+#include "hid/joystick.h"
+#include "net/client.h"
+#include "robots/tank_netsink.h"
+#include "vicon/udp.h"
+
+// Standard C++ includes
+#include <chrono>
+#include <fstream>
+#include <thread>
+
+using namespace BoBRobotics;
+using namespace units::literals;
+using namespace units::time;
+using namespace std::literals;
+
+int
+bob_main(int argc, char **argv)
+{
+    std::string ipAddress;
+    if (argc > 1) {
+        ipAddress = argv[1];
+    } else {
+        std::cout << "IP address [10.0.0.4]: ";
+        std::getline(std::cin, ipAddress);
+        if (ipAddress.empty()) {
+            ipAddress = "10.0.0.4";
+        }
+    }
+
+    // Connect to robot
+    std::cout << "Connecting to robot" << std::endl;
+    Net::Client client(ipAddress);
+    std::cout << "Connected to " << ipAddress << std::endl;
+
+    // Send motor commands to robot
+    Robots::TankNetSink robot(client);
+
+    // Open joystick
+    HID::Joystick joystick;
+    robot.addJoystick(joystick);
+    std::cout << "Opened joystick" << std::endl;
+
+    Stopwatch stopwatch;
+    std::ofstream dataFile;
+    Vicon::UDPClient<Vicon::ObjectDataVelocity> vicon(51001); // For getting robot's position
+
+    // If we're recording, ignore axis movements
+    joystick.addHandler([&stopwatch](HID::JAxis, float) {
+        if (stopwatch.started()) {
+            std::cout << "Ignoring joystick command" << std::endl;
+            return true;
+        } else {
+            return false;
+        }
+    });
+
+    // Toggle testing mode with buttons
+    joystick.addHandler([&](HID::JButton button, bool pressed) {
```

Comment:
Why not just do all the joystick logic by polling? I find the mix of polling calls and callbacks a bit confusing and it splits up the logic between the main loop and the handler (which I'm guessing/hoping is triggered by the ``m_Joystick.update()`` call but might be on a different thread)